src/components/controls/AudioMeter.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+
+export interface IAudioMeterRenderer {
+    (audioMonitor: any): JSX.Element | null | false;
+}
+
+export interface IAudioMeterProps {
+    audioMonitor: any;
+    render?: IAudioMeterRenderer
+};
+
+export class AudioMeter extends React.PureComponent<IAudioMeterProps, {}> {
+    private audioMeter: HTMLDivElement;
+
+    constructor(props: IAudioMeterProps) {
+        super(props);
+
+        this.refMeter = this.refMeter.bind(this);
+    }
+
+    public componentDidMount() {
+        const { audioMonitor } = this.props;
+
+        audioMonitor.on('volume_change', (volume: number) => {
+            if (this.audioMeter) {
+                // NOTE(yunsi): volume ranges from -100 to 0.
+                // TODO(yunsi): Find a better algorithm to convert db to gain.
+                const audioGain = Math.pow(10, (volume / 20));
+                this.audioMeter.style.width = 1000 * audioGain + 'px';
+            }
+        })
+    }
+
+    public refMeter(element: HTMLDivElement) {
+        this.audioMeter = element;
+    }
+
+    public render() {
+        const { render } = this.props;
+
+        if (render) {
+            return render(this.props.audioMonitor)
+        }
+
+        return (
+            <div
```

Comment:
I think we should use a parent container:

```tsx
<div className='audio-meter'>
   <div className='audio-meter__bar'/>
</div>
```

That way `audio-meter__bar` can be a % of `audio-meter` and end-user can specify height of `audio-meter`.

Reply:
Done